src/mainClass.js

```diff
@@ -0,0 +1,107 @@
+import { getData, getAllCategories } from "./source";
+
+export class AllItems {
+
+  setData(searchCategory, searchLinks, displayLinksForChosenCategory, searchChosenLinksOnly) {
```

Comment:
setData not used method, don't create single set and set everything in that.
Create set per property, as in example that I send to you.